Automattic/sensei
# Add theme compatibility for Module page

Partial solution to #2154

Adds theme compatibility for the Module page. A few notes:

- I refactored the request handlers so that each page is handled by a different class. This way `Sensei_Unsupported_Themes` won't get out of hand.

- The "Dummy Post" may not be immediately clear. It's [adapted from WooCommerce](https://github.com/woocommerce/woocommerce/pull/17683). In essence we do the following:
  - Render the Module page to a string using the template.
  - Create a dummy post (of type "page") with enough information to be rendered.
  - Set the dummy post's content to the rendered Module page string.
  - Tell WordPress to render the dummy post using one of the available templates.

- Unlike https://github.com/Automattic/sensei/pull/2155, I opted not to use a Renderer class. It simplified the logic, especially because within the handler, we already know that the globals are all set up for rendering the Module template. In the Renderer, we wouldn't necessarily have that context. If we need to render the Module page in other contexts in the future, we can extract the logic then.

## Still to do


## Testing

- Create a few modules with some Lessons and a description.

- From the Course page, click on one of the Module titles.

- Ensure that the page looks right on themes that support Sensei, and themes that do not support Sensei.

- For themes that do not support Sensei, ensure that:
  - The title is not duplicated.
  - The pagination shows up properly at the bottom.
